### PR TITLE
Add blog post about Game Night

### PR DESCRIPTION
**Pull request by mfrancis95**

### CODE REVIEW

File: _posts/2018-03-29-game-night.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## Introducing Game Night
+
+I created a fully searchable database of our available games called **[Game Night](http://game-night.csh.rit.edu)**. When landing on the home page, the web application lists out all of the games in the database, but since it could be a lot to scroll through, there are ways in which the results can be filtered down.
+
+The site provides a search field at the top to helps users match games in the database that contain a certain keyword or string. If one happens to be familiar regular expressions, they can use those in their queries. For example, searching with the query **b** will match all games that contain a **b** in their name, and searching with the regex query **^b** will match all games that start with a **b**. These queries are case-insensitive.
```
Comment (csssuf) [inline _posts/2018-03-29-game-night.md]: `s/helps/help/`
`s/familiar/familiar with/`

File: _posts/2018-03-29-game-night.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## The Technology Stack
+
+The primary tools I used to develop this web application were Python, Bootstrap 4, Flask and MongoDB. I had never used Flask before, but I was familiar with the similar Express framework used in Node.js. Going from Express to Flask, I absolutely grew to love the Flask framework even more than Express. I found its methodology of using decorators to handle responses to URL routes very elegant. I decided to use MongoDB over the more common MySQL or PostgreSQL for a few reasons:
```
Comment (csssuf) [inline _posts/2018-03-29-game-night.md]: `s/absolutely grew to love/grew to absolutely love/`

File: _posts/2018-03-29-game-night.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+1. It was an opportunity to learn something new.
+2. Since I was only storing games in the database and there were no dependencies on other types of data, I did not need to make use of features offered by typical SQL-based relational database management systems such as foreign keys and joins.
+3. MongoDB stores data in a format almost identical to JSON called BSON, meaning it is simpler to translate that to a format suitable for rendering onto a view or being returned in a RESTful API. Using a SQL-based RDMS typically requires an ORM.
```
Comment (csssuf) [inline _posts/2018-03-29-game-night.md]: `s/onto/into/` or `s/onto/in/`
`s/being returned/returning/`

File: _posts/2018-03-29-game-night.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+Like many other CSH services, **Game Night** is behind OIDC authentication.
+
+## Future Plans
+
+A project like this has a lot of room for augmentation, and I plan on continuing work on it throughout the remainder of my time at RIT. Some features I've already thought about implementing are:
+
+1. A way for users to upload games to the database through the web interface. Currently, only I can upload games.
+2. Including games that are owned by individual CSH members. If the above feature were implemented, then a user uploading a game could specify whether they own it or it belongs to the floor.
```
Comment (csssuf) [inline _posts/2018-03-29-game-night.md]: `s/the floor/floor/`, probably

File: _posts/2018-03-29-game-night.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+author-bio: First Year Computer Science Masters Student at Rochester Institute of Technology
+author-email: maf1144@rit.edu
+author-social:
+    github: https://github.com/mfrancis95
+    linkedin: https://www.linkedin.com/in/mfrancis95/
+---
+
+## Background
+
+In the Fall of 2017, I created a Slack channel for Computer Science House called **#game-night** (formerly known as **#card-games**). The purpose of the channel was to help the organisation's members get together to play many of the fun board and card games owned by CSH. As of the Spring of 2018, the channel has really been taking off. Seeing that rise in popularity, I decided to do more with this neat corner of CSH and develop a web application that would benefit those who frequented the game nights and the organisation as a whole.
```
Comment (jefftaglic) [inline _posts/2018-03-29-game-night.md]: s/organisation/organization/r (twice here)

**Comment by csssuf** [comment]
Make sure you update the date both in the filename and the front matter before merging.

**Pull request merged**